crates/router/src/core/payments/flows/authorize_flow.rs

```diff
@@ -216,6 +216,19 @@ impl Feature<api::Authorize, types::PaymentsAuthorizeData> for types::PaymentsAu
                         self.request.payment_method_type,
                     )
                     .to_payment_failed_response()?;
+
+                if crate::connector::utils::PaymentsAuthorizeRequestData::is_mandate_payment(
+                    &self.request,
```

Comment:
this function checks only whether `setup_mandate_data` is present or `connector_mandate_ids` are present. Are these checks enough to tell whether its a mandate payment @Aprabhat19 @vspecky ?

Reply:
No its not enough, recently we have changed the mandate flow such that we would not create a mandate_id on our end , so this `is_mandate_payment` for every recurring mandate payment would come as false as we won't have a mandate on our end but rather directly with the connector. So we should also add one more check here , saying if `payment_method_info.connector_mandate_details.is_some()` then also its a recurring mandate payment
CC: @vspecky

Reply:
@Aprabhat19 Just the presence of `connector_mandate_details` does not imply a mandate payment right?

Also, if a recurring payment is being done using an MIT (i.e. the connector mandate id is fetched from `payment_method_info.connector_mandate_details`), then routing will populate the `payment_data.mandate_id` field such that the check in the function in question should pass. And since all this happens before router data construction (i.e. before the `authorize_flow` is called), it should be fine afaik. Let me know if I'm missing something.

CC: @Narayanbhat166

Reply:
Nope you are right, my bad I got confused with the fn `is_mandate_payment` that was being called. Also in addition  shouldn't we also check for the presence of the `setup_future_usage : off_session` and `customer_acceptance` , in order to decide if it's the initial mandate payment? 
CC: @vspecky , @Narayanbhat166